omen2/codegen.py

```diff
@@ -0,0 +1,185 @@
+import keyword
+import os
+import sys
+import importlib
+import importlib.util
+import logging as log
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from notanorm import DbTable
+
+
+class CodeGen:
+    def __init__(self, module_path):
+        """Create an omen2 codegen object.
+
+        Args:
+            module_path: packa.module.ClassName
+        """
+        self.path = module_path
+        self.package, self.module, self.class_name = self.parse_class_path(self.path)
+        if self.module == "__main__":
+            self.module, _ = os.path.splitext(
+                os.path.basename(sys.modules["__main__"].__file__)
+            )
+        self.base_cls = self.import_mod()
+        self.model = self.base_cls.model
+
+        # trivially escape all reserved words
+        # if this isn't good enough, you're using some weird db column names
+        for tab in self.model.values():
+            for col in tab.columns:
+                if keyword.iskeyword(col.name):
+                    col.name = col.name + "_"
+
+    @staticmethod
+    def gen_class(out, name, dbtab: "DbTable"):
+        """Generate the derived classes for a single DBTable
+
+        Args:
+            out: file stream
+            name: table name
+            dbtab: table model
+        """
+        # pylint: disable=import-outside-toplevel
+        from omen2.omen import default_type
+
+        print("class " + name + "_row(ObjBase):", file=out)
```

Comment:
extensive comments would be helpful in this method

Reply:
i will clean up coverage and comments in the codegen file.